SS_Activation_OpenMC.py

```diff
@@ -0,0 +1,167 @@
+spectrum_tally = openmc.Tally(name="Flux spectrum")
+# Implementing energy and cell filters for flux spectrum tally
+spectrum_tally.filters = [energy_filter_flux, cell_filter]
+spectrum_tally.scores = ['flux']
+
+# Collecting and exporting tallies to .xml
+tallies = openmc.Tallies([neutron_tally, spectrum_tally])
+tallies.export_to_xml()
+
+model = openmc.model.Model(geometry=geometry,settings=settings)
+#Depletion calculation
+W.depletable = True
+W.volume = 4.0/3.0 * np.pi * (R_2**3 - R_1**3) #volume of W wall material
+operator = openmc.deplete.CoupledOperator(model, normalization_mode='source-rate')
+time_steps = [3e8, 86400, 2.6e6]
+source_rates = [1E+18, 0, 0]
+integrator = openmc.deplete.PredictorIntegrator(operator=operator, timesteps=time_steps, source_rates=source_rates, timestep_units='s')
+integrator.integrate()
+
+#Opening statepoint file to read tallies:
+with openmc.StatePoint('statepoint.10.h5') as sp:
+    fl = sp.get_tally(name="Flux spectrum")
+    nt = sp.get_tally(name="Neutron tally")
+    
+# Get the neutron energies from the energy filter
+energy_filter_fl = fl.filters[0]
+energies_fl = energy_filter_fl.bins[:, 0]
+
+# Get the neutron flux values
+flux = fl.get_values(value='mean').ravel()
+
+#Neutron flux/elastic/absorption tallies:
+tal = nt.get_values(value='mean').ravel()
+print(tal)
+
+Flux_Data = np.c_[energies_fl, flux]
+#Creating an excel file that stores flux data for each energy bin (used as input for ALARA)
+#Dividing by volume to obtain proper units of flux (#/cm^2-s)
+Flux_Data = np.c_[energies_fl, flux/W.volume]
+#ALARA flux inputs go from high energy to low energy
+Flux_Data_ALARA = Flux_Data[::-1]
+FD_CSV = pd.DataFrame(Flux_Data_ALARA, columns=['Energy [eV]', 'Flux [n-cm/sp]'])
+FD_CSV.to_csv('Neutron_Flux.csv', index=False)
+
+Tallies_CSV = pd.DataFrame(tal)
+#Creating a csv file that stores total tally value data
+Tallies_CSV.to_csv('Tally_Values.csv', index=False)
+
+# Depletion results file
+results = openmc.deplete.Results(filename='depletion_results.h5')
+
+# Stable W nuclides present at beginning of operation (will not be plotted)
+stable_nuc = ['W180', 'W182', 'W183', 'W184', 'W186']  
+
+#Store list of nuclides from last timestep as a Materials object
+materials_last = results.export_to_materials(-1)
+# Storing depletion data from 1st material
+mat_dep = materials_last[0]
+# Obtaining the list of nuclides from the results file
+nuc_last = mat_dep.get_nuclides()
+
+# Removing stable W nuclides from list so that they do not appear in the plot
+for j in stable_nuc :
+    nuc_last.remove(j)
+print(nuc_last)
+
+colors = list(mcolors.CSS4_COLORS.keys())
+num_dens= {}
+pair_list = {}
+
+with open(r'Densities_CSV.csv', 'a') as density_file:
+
+    for nuclide in nuc_last:
+        plot_color = random.choice(colors)
+        time, num_dens[nuclide] = results.get_atoms('1', nuclide, nuc_units = 'atom/cm3')
+        print(time, num_dens[nuclide])
+        density_file.write(''.join(map(str, nuclide)) + ',')
```

Comment:
Since this is a single nuclide, this join approach is probably overkill
```suggestion
        density_file.write(f'{nuclide},')
```